src/pages/index.js

```diff
@@ -8,51 +8,164 @@ import Section from "../components/Section.js";
 
-import { Settings, cardData } from "../Utils/Constants.js";
+import { Settings } from "../Utils/Constants.js";
 
 import PopupWithImage from "../components/PopupWithImage.js";
 
 import UserInfo from "../components/UserInfo.js";
 
+import Api from "../components/Api.js";
+
+// API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "9a2a6717-e342-435d-a8a9-13aa291b9ad5",
+    "Content-Type": "application/json",
+  },
+});
+
 /***********
  * Profile *
  ***********/
 
-// Selectors
-
 const profileEditButton = document.querySelector("#profile-edit-button");
-const profileCardModal = document.querySelector("#profile-card-modal");
 const profileEditModal = document.querySelector("#profile-edit-modal");
 const profileEditForm = profileEditModal.querySelector("#editProfileForm");
 
-/********
- * Card *
- ********/
+/*******************
+ * Profile Picture *
+ *******************/
+const profilePictureButton = document.querySelector(".pencil-icon");
+const profilePictureModal = document.querySelector("#profile-picture-modal");
+const profilePictureForm = profilePictureModal.querySelector(
+  "#profile-picture-form"
+);
 
-// Selectors
+/*************
+ * User Info *
+ *************/
 
-const addNewCardButton = document.querySelector(".profile__add-button");
-const addCardForm = profileCardModal.querySelector("#newCardForm");
-const cardSelector = "#card-template";
-const cardListEl = document.querySelector(".cards__list");
+const userInfo = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__picture"
+);
 
-/*****************
- * Preview Image *
- *****************/
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData.name, userData.about, userData.avatar);
+  })
+  .catch((err) => {
+    console.error("Error fetching user info:", err);
+  });
 
-const popupImage = new PopupWithImage("#preview-modal-image");
-popupImage.setEventListeners();
+/***************
+ * ProfileEdit *
+ ***************/
+const editFormValidator = new FormValidator(Settings, profileEditForm);
+editFormValidator.enableValidation();
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+profileEditButton.addEventListener("click", () => {
+  userInfo.getUserInfo();
+  profilePopup.open();
+});
+
+/******************
+ * Profile Submit *
+ ******************/
+
+function handleProfileFormSubmit(formValues) {
+  profilePopup.renderLoading(true);
+
+  api
+    .editUserInfo(formValues)
+    .then((editUserInfo) => {
+      userInfo.setUserInfo(
+        editUserInfo.name,
+        editUserInfo.about,
+        editUserInfo.avatar
+      );
+
+      profilePopup.close();
+      editFormValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.error("Error updating profile:", err);
+    })
+    .finally(() => {
+      profilePopup.renderLoading(false);
+      editFormValidator.disableButton();
```

Comment:
You need to `disable` the button only after **resetting** the form in `then`, otherwise you'll now be able to click the button in case of a server error

**Please, correct it everywhere**